src/executor.rs

```diff
@@ -0,0 +1,150 @@
+                cfg: handler_cfg,
+                context: Context {
+                    evm: evm_context,
+                    external: (),
+                },
+            })
+            .build();
+        run_evm(evm, is_static)
+    }
+}
+
+/// Calls the given evm. This is originally a copy of revm::Evm::transact, but it calls our own output function
+fn run_evm<EXT>(mut evm: Evm<'_, EXT, DB>, is_static: bool) -> PyResult<(ExecutionResult, EvmContext<DB>)> {
+    evm.handler.validation().env(&evm.context.evm.env).map_err(pyerr)?;
+    let initial_gas_spend = evm
+        .handler
+        .validation()
+        .initial_tx_gas(&evm.context.evm.env)
+        .map_err(pyerr)?;
+    evm.handler
+        .validation()
+        .tx_against_state(&mut evm.context)
+        .map_err(pyerr)?;
+
+    let ctx = &mut evm.context;
+    let pre_exec = evm.handler.pre_execution();
+
+    // load access list and beneficiary if needed.
+    pre_exec.load_accounts(ctx).map_err(pyerr)?;
+
+    // load precompiles
+    ctx.evm.set_precompiles(pre_exec.load_precompiles());
+
+    // deduce caller balance with its limit.
+    pre_exec.deduct_caller(ctx).map_err(pyerr)?;
+
+    let gas_limit = ctx.evm.env.tx.gas_limit - initial_gas_spend;
+
+    let exec = evm.handler.execution();
+    // call inner handling of call/create
+    let first_frame_or_result = match ctx.evm.env.tx.transact_to {
+        TransactTo::Call(_) => exec.call(
+            ctx,
+            call_inputs(&ctx, gas_limit, is_static),
+        ).map_err(pyerr)?,
+        TransactTo::Create(_) => exec.create(
+            ctx,
+            CreateInputs::new_boxed(&ctx.evm.env.tx, gas_limit).unwrap(),
+        ).map_err(pyerr)?,
+    };
+
+    // Starts the main running loop.
+    let mut result = match first_frame_or_result {
+        FrameOrResult::Frame(first_frame) => evm.start_the_loop(first_frame).map_err(pyerr)?,
+        FrameOrResult::Result(result) => result,
+    };
+
+    let ctx = &mut evm.context;
+
+    // handle output of call/create calls.
+    evm.handler
+        .execution()
+        .last_frame_return(ctx, &mut result)
+        .map_err(pyerr)?;
+
+    let post_exec = evm.handler.post_execution();
+    // Reimburse the caller
+    post_exec.reimburse_caller(ctx, result.gas()).map_err(pyerr)?;
+    // Reward beneficiary
+    post_exec.reward_beneficiary(ctx, result.gas()).map_err(pyerr)?;
+    // Returns output of transaction.
+    Ok((output(ctx, result)?, evm.context.evm))
+}
+
+fn call_inputs<EXT>(ctx: &&mut Context<EXT, DB>, gas_limit: u64, is_static: bool) -> Box<CallInputs> {
+    let mut inputs = CallInputs::new_boxed(&ctx.evm.env.tx, gas_limit).unwrap();
+    inputs.is_static = is_static;
+    inputs
+}
+
+/// Returns the output of the transaction.
+/// This is mostly copied from revm::handler::mainnet::post_execution::output
+/// However, we removed the journal finalization to keep the transaction open.
```

Comment:
This can be good feature set for Revm to have journaling open across transactions. Would need additional work inside JournalState to support it.

There are few `unwraps()` that could panic if revert is done on unknown State: https://github.com/bluealloy/revm/blob/main/crates/revm/src/journaled_state.rs

Reply:
This solution seems to be working in our tests. Or are there other issues I'm overseeing?